apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/ValidatorApp.scala

```diff
@@ -183,6 +184,36 @@ class ValidatorApp(
     }
   } yield ()
 
+  private def persistScanUrlListBuilder(
+      validatorConfigProvider: ValidatorConfigProvider
+  )(implicit traceContext: TraceContext): Seq[(String, String)] => Future[Unit] = {
+
+    (connections: Seq[(String, String)]) =>
+      {
+        val internalConfigs: Seq[ScanUrlInternalConfig] = connections.map { case (svName, url) =>
+          ScanUrlInternalConfig(
+            svName = svName,
+            url = url,
+          )
+        }
+        validatorConfigProvider.setScanUrlInternalConfig(internalConfigs)
+      }
+  }
+
+  private def getPersistedScanList(
+      validatorConfigProvider: ValidatorConfigProvider
+  )(implicit traceContext: TraceContext): Future[Option[List[(String, String)]]] = {
+
+    val optionTConfig = validatorConfigProvider.getScanUrlInternalConfig()
+
+    optionTConfig.map { internalConfigs =>
+      internalConfigs.map { internalConfig =>
+        (internalConfig.svName, internalConfig.url)
+      }.toList
+    }.value
+
+  }
```

Comment:
Lets move this into a `ValidatorScanConnection` class where we can locate all the code related to scan connections